components/kyma-environment-broker/internal/ias/bundle.go

```diff
@@ -124,22 +128,51 @@ func (b *ServiceProviderBundle) DeleteServiceProvider() error {
 	return nil
 }
 
+func (b *ServiceProviderBundle) configureServiceProviderOIDCType(redirectURI string) error {
+	iasType := OIDCType{
+		ServiceProviderName: b.serviceProviderDNS,
+		SsoType:             b.serviceProviderParams.ssoType,
+		OpenIDConnectConfig: OpenIDConnectConfig{
+			RedirectURIs: []string{redirectURI},
+		},
+	}
+
+	return b.client.SetOIDCConfiguration(b.serviceProvider.ID, iasType)
+}
+
+func (b *ServiceProviderBundle) configureServiceProviderSAMLType(redirectURI string) error {
+	iasType := SAMLType{
+		ServiceProviderName: b.serviceProviderDNS,
+		ACSEndpoints: []ACSEndpoint{
+			ACSEndpoint{
+				Location:  redirectURI,
+				Index:     0,
+				IsDefault: "true",
+			},
+		},
+	}
+
+	return b.client.SetSAMLConfiguration(b.serviceProvider.ID, iasType)
+}
+
 // ConfigureServiceProviderType sets SSO type, name and URLs based on provided URL for ServiceProvider
 func (b *ServiceProviderBundle) ConfigureServiceProviderType(consolePath string) error {
 	u, err := url.ParseRequestURI(consolePath)
 	if err != nil {
 		return errors.Wrap(err, "while parsing path for IAS Type")
 	}
-	path := strings.Replace(u.Host, "console.", "grafana.", 1)
+	b.serviceProviderDNS = strings.Replace(u.Host, "console.", fmt.Sprintf("%s.", b.serviceProviderParams.Domain), 1)
```

Comment:
why does it struct parameter if it is used only in this method? It is not better to pass the DNS(name) as an argument?

Reply:
Removed this, that's a remnant of a stupid idea which didn't make it's way to the PR